### DISCUSSION TITLE
Error 500 on fresh linux install on vmware ubuntu 20.x vm

### DISCUSSION THREAD

**Discussion by mihir2780** [JanssenProject/jans#1309] [Q&A]
Hello, I have configured jans setup on ubuntu 20.X on my vmware workstation.I am getting this type of error for both of the modules,
when I try to access menu driven interface. Any solution on the below problem mentioned? 

root@ub:~# /opt/jans/jans-cli/scim-cli.py
An Unhandled error raised: (500)
Reason: Server Error
HTTP response headers: HTTPHeaderDict({'Date': 'Mon, 09 May 2022 07:24:37 GMT', 'Server': 'Apache/2.4.41 (Ubuntu)', 'X-Xss-Protection': '1; mode=block', 'X-Content-Type-Options': 'nosniff', 'Strict-Transport-Security': 'max-age=31536000; includeSubDomains', 'Expires': 'Thu, 01 Jan 1970 00:00:00 GMT', 'Cache-Control': 'no-store', 'Pragma': 'no-cache', 'Content-Length': '0', 'Set-Cookie': 'X-Correlation-Id=242e84d4-beaa-45e5-9b73-f2b08dd7c5b9; Secure; HttpOnly;HttpOnly', 'Connection': 'close'})

Error is logged to /opt/jans/jans-cli/error.log
root@ub:~# /opt/jans/jans-cli/config-cli.py
An Unhandled error raised: (500)
Reason: Server Error
HTTP response headers: HTTPHeaderDict({'Date': 'Mon, 09 May 2022 07:24:47 GMT', 'Server': 'Apache/2.4.41 (Ubuntu)', 'X-Xss-Protection': '1; mode=block', 'X-Content-Type-Options': 'nosniff', 'Strict-Transport-Security': 'max-age=31536000; includeSubDomains', 'Expires': 'Thu, 01 Jan 1970 00:00:00 GMT', 'Cache-Control': 'no-store', 'Pragma': 'no-cache', 'Content-Length': '0', 'Set-Cookie': 'X-Correlation-Id=c0da5066-c184-4a24-9b9d-f30da7d81f86; Secure; HttpOnly;HttpOnly', 'Connection': 'close'})

Error is logged to /opt/jans/jans-cli/error.log

**Comment by nynymike** [comment]
@manojs1978  can you try to replicate?

**Reply by manojs1978** [reply]
Hi @mihir2780 ,
could you please share more details.
 which steps you have followed to install jans on ubuntu?

**Reply by mihir2780** [reply]
Hello, @manojs1978

I have followed this URL to install jans on Ubuntu(Version 20.X)for my VMware Workstation - 

[https://github.com/JanssenProject/jans/wiki/Installation-Guide#dynamic-download](url)

In the above link I have followed the section named Ubuntu 20.

**Reply by manojs1978** [reply]
Hi @mihir2780 ,
i just installed jans using steps in above mentioned url.
and able to configure and access CLI tool
![image](https://user-images.githubusercontent.com/5729240/168727097-e4b1ec7d-4774-4f3d-bac5-81869bc3bbb1.png)
![image](https://user-images.githubusercontent.com/5729240/168727289-35585347-07d7-4f9c-bf0f-3762b002e859.png)



could you please check it again by reinstalling janssen on new VM?
let us know if still facing same issue.

Thanks

**Reply by mihir2780** [reply]
Hello @manojs1978 ,
I have reinstalled janssen on VM couple of times.

**root@ub:~# /opt/jans/jans-cli/scim-cli.py
An Unhandled error raised: (500)
Reason: Server Error
HTTP response headers: HTTPHeaderDict({'Date': 'Mon, 09 May 2022 07:24:37 GMT', 'Server': 'Apache/2.4.41 (Ubuntu)', 'X-Xss-Protection': '1; mode=block', 'X-Content-Type-Options': 'nosniff', 'Strict-Transport-Security': 'max-age=31536000; includeSubDomains', 'Expires': 'Thu, 01 Jan 1970 00:00:00 GMT', 'Cache-Control': 'no-store', 'Pragma': 'no-cache', 'Content-Length': '0', 'Set-Cookie': 'X-Correlation-Id=242e84d4-beaa-45e5-9b73-f2b08dd7c5b9; Secure; HttpOnly;HttpOnly', 'Connection': 'close'})

Error is logged to /opt/jans/jans-cli/error.log
root@ub:~# /opt/jans/jans-cli/config-cli.py
An Unhandled error raised: (500)
Reason: Server Error
HTTP response headers: HTTPHeaderDict({'Date': 'Mon, 09 May 2022 07:24:47 GMT', 'Server': 'Apache/2.4.41 (Ubuntu)', 'X-Xss-Protection': '1; mode=block', 'X-Content-Type-Options': 'nosniff', 'Strict-Transport-Security': 'max-age=31536000; includeSubDomains', 'Expires': 'Thu, 01 Jan 1970 00:00:00 GMT', 'Cache-Control': 'no-store', 'Pragma': 'no-cache', 'Content-Length': '0', 'Set-Cookie': 'X-Correlation-Id=c0da5066-c184-4a24-9b9d-f30da7d81f86; Secure; HttpOnly;HttpOnly', 'Connection': 'close'})

Error is logged to /opt/jans/jans-cli/error.log**

The above error which I also mentioned earlier in the discussion chain is observed after few days after when  the installation is done.
After reinstalling janssen again on VM, after few days it won't work.

Regards,
Mihir

**Reply by ossdhaval** [reply]
@mihir2780 Just wanted to confirm that at the time of reinstall, you removed the old `install.py` and used freshly downloaded `install.py`?
@mbaser would using an old `install.py` can cause issue here?

**Reply by mihir2780** [reply]
Hello @ossdhaval,
Using VMware workstation, we take snapshots of all VM. 
Using that, we are able revert to required snapshots. So, if our installation fails 
We revert it back to the state where nothing is installed and start following the installation steps again. 
So, its always freshly used **setup.py** in my case.

Regards,
Mihir

**Comment by ossdhaval** [comment]
Closing as the discussed issue is pretty old and maybe not relevant anymore.
